data/beaconrestapi/src/main/java/tech/pegasys/teku/beaconrestapi/EthereumTypes.java

```diff
@@ -31,4 +39,17 @@ public class EthereumTypes {
           .description("`BLSSignature Hex` BLS12-381 signature for the current epoch.")
           .format("byte")
           .build();
+
+  public static <T extends SszData> ResponseContentTypeDefinition<T> sszResponseType(
+      final Function<T, SpecMilestone> milestoneSelector) {
+    return new OctetStreamResponseContentTypeDefinition<>(
+        SszData::sszSerialize,
+        value ->
+            Map.of(
+                RestApiConstants.HEADER_CONSENSUS_VERSION,
+                Version.fromMilestone(milestoneSelector.apply(value)).name(),
+                RestApiConstants.HEADER_CONTENT_DISPOSITION,
+                String.format(
+                    "filename=\"%s.ssz\"", value.hashTreeRoot().toUnprefixedHexString())));
```

Comment:
i know we do this currently, wondering if it is worth putting `className_hash.ssz` if it's going to be used for everything now rather than just state?

Reply:
I was originally thinking `getSchema().getName()` so that's probably worth doing yeah.

Reply:
Turns out we don't always have a name for SszData but we now include it when we do.  And for the things we're likely to be returning (states and blocks) we should always have a name.